File: app/src/main/java/com/djrausch/billtracker/util/NotificationID.java
package com.djrausch.billtracker.util;

import android.content.Context;
import android.content.SharedPreferences;

import java.util.concurrent.atomic.AtomicInteger;

/**
 * Created by white on 7/25/2016.
 */
public class NotificationID {
    private static final String ATOMIC_ID = "atomicId";
    private Context context;
    private AtomicInteger atomicInt;
    private SharedPreferences preferences;

    public NotificationID(Context context) {
        this.context = context;

        preferences = context.getSharedPreferences("bill_tracker_internal_settings", Context.MODE_PRIVATE);
        int a = preferences.getInt(ATOMIC_ID, 0);
        atomicInt = new AtomicInteger(a);
    }

    public int get() {
        int atomicIntNew = atomicInt.incrementAndGet();
        SharedPreferences.Editor editor = preferences.edit();
        editor.putInt(ATOMIC_ID, atomicIntNew);
        editor.apply();
        return atomicIntNew;
    }
}

File: app/src/main/java/com/djrausch/billtracker/util/NotificationID.kt
package com.djrausch.billtracker.util

import android.content.Context
import android.content.SharedPreferences

import java.util.concurrent.atomic.AtomicInteger

class NotificationID(context: Context) {
    private var atomicInt: AtomicInteger? = null
    private val preferences: SharedPreferences

    init {
        preferences = context.getSharedPreferences(
                "bill_tracker_internal_settings", Context.MODE_PRIVATE)
        val a = preferences.getInt("atomicId", 0)
        atomicInt = AtomicInteger(a)
    }

    fun get(): Int {
        val atomicIntNew = atomicInt!!.incrementAndGet()
        val editor = preferences.edit()
        editor.putInt("atomicId", atomicIntNew)
        editor.apply()
        return atomicIntNew
    }
}